Source/cielim/Private/RenderingFunctionsLibrary.cpp

```diff
@@ -26,3 +28,79 @@ void URenderingFunctionsLibrary::ApplyReadNoise()
 {
 	
 }
+
+FString URenderingFunctionsLibrary::ApplyDarkCurrentNoise(FString Filepath, double MaxSigma, double MinSigma, FVector SunPosition, FVector SpacecraftPosition, FVector SpacecraftDirection)
+{
+
+	//Read Image
+	FString Filepath_Absolute = IFileManager::Get().ConvertToAbsolutePathForExternalAppForRead(*Filepath);
+	std::string Filepath_Absolute_String = TCHAR_TO_UTF8(*Filepath_Absolute);
+	cv::Mat Image = cv::imread(Filepath_Absolute_String);
+
+	//Init Resulting image
+	cv::Mat ResultImage = cv::Mat(Image.rows, Image.cols, Image.type());
+
+	//Calculate Influence of Sun on noise
+	FVector DirectionVector = SunPosition - SpacecraftPosition;
+	DirectionVector.Normalize();
+	double DirectionDotProduct = SpacecraftDirection.Dot(DirectionVector);
+	DirectionDotProduct = (DirectionDotProduct + 1.0) / 2.0;
+
+	//Calculate noise
+	double Sigma =  DirectionDotProduct * (MaxSigma - MinSigma) + MinSigma;
+
+	//Print Sigma
+	UE_LOG(LogCielim, Warning, TEXT("Sigma: %s"), *FString::SanitizeFloat(Sigma))
+	
+	//Return original image
+	if( Sigma < 0 )
+	{
+		//No noise added
+		ResultImage = Image;
+		
+		//Save image
+		FString ResultFilepath = FPaths::ProjectDir();
+		ResultFilepath.Append("Result_Images/");
+		ResultFilepath.Append("DarkCurrentNoise.jpg");
+	
+		std::string ResultFilepath_String = TCHAR_TO_UTF8(*ResultFilepath);
+		cv::imwrite(ResultFilepath_String, ResultImage);
+	
+		return ResultFilepath;
+	}
+	
+	//Init and create noise matrix
+	cv::Mat GaussianNoise = cv::Mat::zeros(Image.rows, Image.cols, Image.type());
+	cv::randn(GaussianNoise, 0, Sigma);
+
+	//Apply Noise
+	ResultImage = Image + GaussianNoise;
+
+	//Clamp values to [0,255]
+	//Separate color channels
+	std::vector<cv::Mat> DifferentColorChannels;
+	cv::split(ResultImage, DifferentColorChannels);
+
+	//Init LowerBound and UpperBound matrices
+	cv::Mat LowerBoundMatrix = cv::Mat::zeros(Image.rows, Image.cols, DifferentColorChannels[0].type());
+	cv::Mat UpperBoundMatrix = cv::Mat::ones(Image.rows, Image.cols, DifferentColorChannels[0].type()) * 255;
+
+	//Clamp
+	for(int ColorChannel =0; ColorChannel < 3; ColorChannel++)
+	{
+		cv::min(cv::max(DifferentColorChannels[ColorChannel], LowerBoundMatrix), UpperBoundMatrix, DifferentColorChannels[ColorChannel]);
```

Comment:
:large_orange_diamond: Because this line extends over 120 characters, you can drop each parameter onto a new line.

Reply:
Put each parameter onto a new line!